api/v1alpha1/hosted_controlplane.go

```diff
@@ -195,6 +195,11 @@ type HostedControlPlaneStatus struct {
 	// +kubebuilder:validation:Optional
 	ControlPlaneEndpoint APIEndpoint `json:"controlPlaneEndpoint,omitempty"`
 
+	// OAuthURL contains the URL where OAuth can be accessed for the
+	// HostedCluster. This is populated after the infrastructure is ready.
+	// +kubebuilder:validation:Optional
+	OAuthURL string `json:"oauthURL,omitempty"`
```

Comment:
This is the URL part of the callback url and only needed to set up an IDP, is that correct? If so, would it make sense to call this `OauthCallBackURL` and add the url path element to simplify usage?

Reply:
So we wouldn't be able to give you the full oauth callback URL because it requires the identity provider name to be added in the path:
(see https://docs.openshift.com/container-platform/4.10/authentication/identity_providers/configuring-github-identity-provider.html) 
If we give you something like:
https://oauth-openshift.apps.thecluster.example.com/oauth2callback
you still need to append to it the name of your configured provider:
https://oauth-openshift.apps.thecluster.example.com/oauth2callback/github

imo calling it OAuthCallbackURL could be misleading you to think that's all you need.
or we could add the [identity-provider-name] placeholder to it as well, wdyt?

Reply:
Adding a placeholder seems like a nice idea to me